sdcflows/utils/tests/test_tools.py

```diff
@@ -23,11 +23,50 @@
 """Test EPI manipulation routines."""
 import numpy as np
 import nibabel as nb
-from ..tools import brain_masker
+from sdcflows.utils.tools import brain_masker, deoblique_and_zooms
 
 
 def test_epi_mask(tmpdir, testdata_dir):
     """Check mask algorithm."""
     tmpdir.chdir()
     mask = brain_masker(testdata_dir / "epi.nii.gz")[-1]
     assert abs(np.asanyarray(nb.load(mask).dataobj).sum() - 166476) < 10
+
+
+def test_deoblique_and_zooms():
+    """Check deoblique and denser."""
+
+    # Generate an example reference image
+    ref_data = np.zeros((20, 30, 40), dtype=np.float32)
+    ref_affine = np.eye(4)
+    ref_affine[:3, :3] = np.diag([2, 3, 4])  # Set zooms to (2, 3, 4)
+    ref_img = nb.Nifti1Image(ref_data, ref_affine)
+
+    # Generate an example oblique image
+    ob_data = np.zeros((25, 35, 45), dtype=np.float32)
+    ob_affine = np.eye(4)
+
+    # Rotate 90 degrees around x-axis
+    ob_affine[:3, :3] = np.array([[0, 1, 0], [0, 0, 1], [1, 0, 0]])
+    ob_img = nb.Nifti1Image(ob_data, ob_affine)
+
+    # Call function with default parameters
+    out_img = deoblique_and_zooms(ref_img, ob_img)
+
+    # Check output shape and zooms
+    assert out_img.shape == (26, 36, 46)
+    assert np.allclose(out_img.header.get_zooms()[:3], (0.5, 1.0, 1.0))
```

Comment:
Some explanation would help. Why are the zooms 0.5x1x1? If we're upsampling by a factor of 4, shouldn't it be 0.5x0.75x1.0? And if we're padding by 1 in each direction, shouldn't the shape be (27, 37, 47)?

And what impact does the 90deg rotation have? Should it be 0.5x1.0x0.75 and (27, 47, 37)?

Reply:
Sorry, I was still on it. I can give you a heads up when the test is ready.